pkg/controller/federatedcluster/clusterclient.go

```diff
@@ -85,30 +87,35 @@ func (self *ClusterClient) GetClusterHealthStatus() *fedv1a1.FederatedClusterSta
 		LastProbeTime:      currentTime,
 		LastTransitionTime: currentTime,
 	}
-	newNodeOfflineCondition := fedv1a1.ClusterCondition{
+	newClusterOfflineCondition := fedv1a1.ClusterCondition{
 		Type:               fedcommon.ClusterOffline,
 		Status:             corev1.ConditionTrue,
 		Reason:             "ClusterNotReachable",
 		Message:            "cluster is not reachable",
 		LastProbeTime:      currentTime,
 		LastTransitionTime: currentTime,
 	}
-	newNodeNotOfflineCondition := fedv1a1.ClusterCondition{
+	newClusterNotOfflineCondition := fedv1a1.ClusterCondition{
 		Type:               fedcommon.ClusterOffline,
 		Status:             corev1.ConditionFalse,
 		Reason:             "ClusterReachable",
 		Message:            "cluster is reachable",
 		LastProbeTime:      currentTime,
 		LastTransitionTime: currentTime,
 	}
-	body, err := self.kubeClient.DiscoveryClient.RESTClient().Get().AbsPath("/healthz").Do().Raw()
-	if err != nil {
-		clusterStatus.Conditions = append(clusterStatus.Conditions, newNodeOfflineCondition)
-	} else {
-		if !strings.EqualFold(string(body), "ok") {
-			clusterStatus.Conditions = append(clusterStatus.Conditions, newClusterNotReadyCondition, newNodeNotOfflineCondition)
+
+	for i := 0; i < maxHealthCheckRetries; i++ {
+		clusterStatus.Conditions = clusterStatus.Conditions[:0]
+		body, err := self.kubeClient.DiscoveryClient.RESTClient().Get().AbsPath("/healthz").Do().Raw()
+		if err != nil {
```

Comment:
probably user could configure wrong cluster credentials, which might return error. Preserving the old behaviour for now

Reply:
I think it would be preferable to not maintain the old behavior given that it hides details that is likely to be useful for troubleshooting.  If you're not intending to fix the problem in this PR I can raise a fix separately.

Reply:
Have added runtime.HandleError. hope this is as you expected.